sdk/state/open.go

```diff
@@ -224,7 +229,11 @@ func (c *Channel) ProposeOpen(p OpenParams) (OpenAgreement, error) {
 
 func (c *Channel) validateOpen(m OpenAgreement) error {
 	// if the channel is already open, error.
-	if c.openAgreement.isFull() {
+	cs, err := c.State()
+	if err != nil {
+		return fmt.Errorf("getting channel state: %w", err)
+	}
+	if cs >= StateOpen {
```

Comment:
I think this is better than `if cs != StateNone`, in the case the channel errored on the first attempt to open, so they should be allowed to try again